expan/core/early_stopping.py

```diff
@@ -0,0 +1,228 @@
+			of evaluation, e.g. the share of the maximum sample size
+		alpha: type-I error rate
+
+	Returns:
+		float: redistributed alpha value at the time point with the given 
+			   information fraction
+	"""
+	return (1-norm.cdf(norm.ppf(1-alpha/2)/np.sqrt(information_fraction)))*2
+
+
+def group_sequential(x, 
+					 y, 
+					 spending_function='obrien_fleming', 
+					 information_fraction=1, 
+					 alpha=0.05, 
+					 cap=8):
+	"""
+	Group sequential method to determine whether to stop early or not.
+
+	Args:
+		x (array_like): sample of a treatment group
+		y (array_like): sample of a control group
+		spending_function: name of the alpha spending function, currently
+			supports: 'obrien_fleming'
+		information_fraction: share of the information amount at the point 
+			of evaluation, e.g. the share of the maximum sample size
+		alpha: type-I error rate
+		cap: upper bound of the adapted z-score
+
+	Returns:
+		tuple: 
+			- stop label
+			- effect size (delta)
+			- confidence interval of delta
+			- sample size of x
+			- sample size of y
+			- absolute mean of x
+			- absolute mean of y
+	"""
+	# Checking if data was provided
+	if x is None or y is None:
+		raise ValueError('Please provide two non-None samples.')
+
+	# Coercing missing values to right format
+	_x = np.array(x, dtype=float) 
+	_y = np.array(y, dtype=float) 
+
+	# if scalar, assume equal spacing between the intervals
+	#if not isinstance(information_fraction, list): 
+	#	fraction = np.linspace(0,1,information_fraction+1)[1:]
+	#else:
+	#	fraction = information_fraction
+
+	# alpha spending function
+	if spending_function in ('obrien_fleming'):
+		func = eval(spending_function)
+	else:
+		raise NotImplementedError
+	alpha_new = func(information_fraction, alpha=alpha)
+
+	# calculate the z-score bound
+	bound = norm.ppf(1-alpha_new/2)
+	# replace potential inf with an upper bound
+	if bound == np.inf:
+		bound = cap
+
+	mu_x = np.nanmean(_x)
+	mu_y = np.nanmean(_y)
+	sigma_x = np.nanstd(_x)
+	sigma_y = np.nanstd(_y)
+	n_x = statx.sample_size(_x)
+	n_y = statx.sample_size(_y)
+	z = (mu_x-mu_y) / np.sqrt(sigma_x**2/n_x+sigma_y**2/n_y)
+    
+	if z > bound or z < -bound:
+		stop = 1
+	else:
+		stop = 0
+
+	interval = statx.normal_difference(mu_x,sigma_x,n_x,mu_y,sigma_y,n_y,[alpha_new*100/2,100-alpha_new*100/2])
+
+	return stop, mu_x-mu_y, interval, n_x, n_y, mu_x, mu_y
+
+
+def HDI_from_MCMC(posterior_samples, credible_mass=0.95):
+    # Computes highest density interval from a sample of representative values,
+    # estimated as the shortest credible interval
+    # Takes Arguments posterior_samples (samples from posterior) and credible mass (normally .95)
+    # http://stackoverflow.com/questions/22284502/highest-posterior-density-region-and-central-credible-region
+    sorted_points = sorted(posterior_samples)
+    ciIdxInc = np.ceil(credible_mass * len(sorted_points)).astype('int')
+    nCIs = len(sorted_points) - ciIdxInc
+    ciWidth = [0]*nCIs
```

Comment:
switch to np.array instead of lists and get rid of the for loop below in favour of array sections?